server/src/http/mod.rs

```diff
@@ -146,9 +147,16 @@ pub fn run_http_server(
 
     // Spawn a tokio task to monitor for quit and to shutdown the http server
     thread::spawn(move || {
-        tokio::run(quit_task.map(move |_| {
-            info!("HTTP server will be stopped");
-            let _ = addr.send(server::StopServer { graceful: false });
-        }));
+        if !single_threaded_executor {
+            tokio::runtime::run(quit_task.map(move |_| {
```

Comment:
Can you assign this closure to a variable, e.g. stop_server_task because the if/else are running the same closure.

Reply:
Yeah, agree that would be better 👍🏻